xdsl/ir.py

```diff
@@ -918,12 +918,12 @@ def is_structurally_equivalent(
         if self.parent and other.parent and context.get(self.parent) != other.parent:
```

Comment:
Why doesn't your test run into this line? Both ops don't have parents, right? So self.parent and other.parent should be None :thinking:

Reply:
Yeah, but `None` is converted to `False`, which is what this is checking.
(It's equivalent to `if self.parent is not None and ...`

Reply:
Ah, I see, guess I was confused...